### PR TITLE
Аdd db size to userData

### PR DESCRIPTION
**Pull request by munakoiso**
- fix race in tarBall.size
- some refactoring in stream push
- add dataSize and dataTarSize to userData
- fix flap (wait while pg not ready)

### CODE REVIEW

File: internal/databases/mongo/stream_push_handler.go [hunk 1]
```diff
@@ -3,29 +3,17 @@ package mongo
...
 	"io"
-	"io/ioutil"
 
 	"github.com/tinsane/tracelog"
 	"github.com/wal-g/wal-g/utility"
 )
 
 func HandleStreamPush(uploader *Uploader, command []string) {
-	waitFunc, stream, errorStream := internal.StartCommand(command)
+	waitAndFatalOnError, stream := internal.StartCommand(command)
```
Comment (reshke) [inline internal/databases/mongo/stream_push_handler.go]: nice

File: internal/start_command.go [hunk 2]
```diff
@@ -15,5 +16,18 @@ func StartCommand(command []string) (waitFunc func() error, stdout, stderr io.Re
 	err = c.Start()
 	waitFuncResult := c.Wait
 	tracelog.ErrorLogger.FatalOnError(err)
-	return waitFuncResult, stdoutResult, stderrResult
+	return func() {
```
Comment (reshke) [inline internal/start_command.go]: here we lose code readability. I think this function is too big to declare it and call it in one place of the code

File: internal/uploader.go [hunk 3]
```diff
@@ -22,6 +22,7 @@ type Uploader struct {
 	deltaFileManager     *DeltaFileManager
 	archiveStatusManager DataFolder
 	Failed               atomic.Value
+	tarSize              *int64
```
Comment (reshke) [inline internal/uploader.go]: is there any reason why we use a pointer, is it possible to simply pass by value?

File: internal/start_command.go (hunk 2 above)
Comment (munakoiso) [reply internal/start_command.go]: I don't know other ways to declare this function with variables from StartCommand

File: internal/uploader.go (hunk 3 above)
Comment (munakoiso) [reply internal/uploader.go]: It's not possible because of races.

File: internal/backup_push_handler.go [hunk 4]
```diff
@@ -171,7 +173,12 @@ func HandleBackupPush(uploader *Uploader, archiveDirectory string, isPermanent b
 
 	currentBackupSentinelDto.setFiles(bundle.GetFiles())
 	currentBackupSentinelDto.BackupFinishLSN = &finishLsn
-	currentBackupSentinelDto.UserData = GetSentinelUserData()
+	userData := map[string]interface{}{"backupSize": dataSize, "backupTarSize": atomic.LoadInt64(uploader.tarSize)}
```
Comment (x4m) [inline internal/backup_push_handler.go]: Please create typed field in dto classes

File: internal/storage_tar_ball.go [hunk 5]
```diff
@@ -15,7 +16,7 @@ import (
 type StorageTarBall struct {
 	backupName  string
 	partNumber  int
-	size        int64
+	size        *int64
```
Comment (x4m) [inline internal/storage_tar_ball.go]: technically, this is not size of tarball. This is a size of all tarballs.
I'm not sure this is a good place to store size of all tarballs, but let's leave it aside, just rename it correctly.

**Pull request merged**